swap/htlc.go

```diff
@@ -512,3 +569,147 @@ func (h *HtlcScriptV2) MaxTimeoutWitnessSize() int {
 func (h *HtlcScriptV2) SuccessSequence() uint32 {
 	return 1
 }
+
+// HtlcScriptV2 encapsulates the htlc v2 script.
+type HtlcScriptV3 struct {
+	timeoutScript  []byte
+	claimScript    []byte
+	taprootKey     *secp.PublicKey
+	internalPubKey *secp.PublicKey
+	senderKey      [33]byte
+}
+
+func newHTLCScriptV3(
+	cltvExpiry int32, senderHtlcKey, receiverHtlcKey [33]byte, swapHash lntypes.Hash) (*HtlcScriptV3, error) {
+
+	/*
+		CLAIM PATH
+
+		<reciever_key> OP_CHECKSIGVERIFY OP_SIZE 20 OP_EQUALVERIFY OP_RIPEMD160 <hash> OP_EQUALVERIFY 1 OP_CHECKSEQUENCEVERIFY
+	*/
+	builder := txscript.NewScriptBuilder()
+
+	builder.AddData(receiverHtlcKey[:])
+	builder.AddOp(txscript.OP_CHECKSIGVERIFY)
+	builder.AddOp(txscript.OP_SIZE)
+	builder.AddInt64(32)
+	builder.AddOp(txscript.OP_EQUALVERIFY)
+	builder.AddOp(txscript.OP_HASH160)
+	builder.AddData(input.Ripemd160H(swapHash[:]))
+	builder.AddOp(txscript.OP_EQUALVERIFY)
+	builder.AddInt64(1)
+	builder.AddOp(txscript.OP_CHECKSEQUENCEVERIFY)
+	claimPathScript, err := builder.Script()
+
+	if err != nil {
+		return nil, err
+	}
+
+	/*
+		TIMEOUT PATH
+
+		<timeout_key> OP_CHECKSIGVERIFY <timeout height> OP_CHECKLOCKTIMEVERIFY
+	*/
+	builder = txscript.NewScriptBuilder()
+	builder.AddData(senderHtlcKey[:])
+	builder.AddOp(txscript.OP_CHECKSIGVERIFY)
+	builder.AddInt64(int64(cltvExpiry))
+	builder.AddOp(txscript.OP_CHECKLOCKTIMEVERIFY)
+	timeoutPathScript, err := builder.Script()
+
+	if err != nil {
+		return nil, err
+	}
+
+	randomPub, _ := hex.DecodeString(
```

Comment:
If we're going to deploy these chagnes incrementally (ie, do tapscript then musig keyspend) this is going to need to be deterministic / communicated between client/server so that each party knows the other doesn't know the privkey (and can just claim the htlc). 

Probably not worth the effort, so would just add a clarifying comment that this is a dummy value that *should not* be used in prod.